realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -0,0 +1,175 @@
+    public void testMoveDown() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog1 = owner.getDogs().get(1);
+        owner.getDogs().move(1, 0);
+
+        assertEquals(0, owner.getDogs().indexOf(dog1));
+    }
+
+    // Test move where oldPosition < newPosition
+    public void testMoveUp() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog5 = owner.getDogs().get(5);
+        owner.getDogs().move(5, 6); // This doesn't do anything as index 5 is now empty so the index's above gets shifted to the left.
+
+        assertEquals(10, owner.getDogs().size());
+        assertEquals(5, owner.getDogs().indexOf(dog5));
+    }
+
+    public void testMoveOutOfBoundsLowerThrows() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        try {
+            owner.getDogs().move(1, -1);
+        } catch (IndexOutOfBoundsException e) {
+            return;
+        }
+
+        fail("Indexes < 0 should throw an exception");
+    }
+
+    public void testMoveOutOfBoundsHigherThrows() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        try {
+            owner.getDogs().move(9, 10);
+        } catch (IndexOutOfBoundsException e) {
+            return;
+        }
+
+        fail("Indexes >= size() should throw an exception");
+    }
+
+    public void testAddObject() {
+        testRealm.beginTransaction();
+        testRealm.clear(Owner.class);
+        Owner owner = testRealm.createObject(Owner.class);
+        Dog dog = testRealm.createObject(Dog.class);
+        owner.getDogs().add(dog);
+        testRealm.commitTransaction();
+
+        assertEquals(1, testRealm.where(Owner.class).findFirst().getDogs().size());
+    }
+
+    public void testAddObject_NullThrows() {
+        testRealm.beginTransaction();
+        Owner owner = testRealm.createObject(Owner.class);
+        try {
+            owner.getDogs().add(null);
+        } catch (NullPointerException e) {
+            return;
+        } finally {
+            testRealm.cancelTransaction();
+        }
+
+        fail("Adding null values is not supported");
+    }
+
+    public void testSize() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        assertEquals(10, owner.getDogs().size());
+    }
+
+    public void testGetObjects() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> dogs = owner.getDogs();
+
+        assertNotNull(dogs);
+        assertEquals("Dog 1", dogs.get(1).getName());
+    }
+
+    public void testGetFirstObject() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog = owner.getDogs().first();
+
+        assertEquals("Dog 0", dog.getName());
+    }
+
+    public void testGetLastObject() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog = owner.getDogs().last();
+
+        assertEquals("Dog 9", dog.getName());
```

Comment:
The value `"Dog 9"` depends on the value of `TEST_OBJECTS`, and there are two cases-

The first case it `TEST_OBJECTS` is greater or equal to 10. If you compute the number of digits in `TEST_OBJECT`, you know the number of 9s. 

If `TEST_OBJECTS` is less than 10, then the value can be constructed using value of `TEST_OBJECTS` directly.